src/corenetwork/lteip/IP2lte.cc

```diff
@@ -154,18 +192,17 @@ void IP2lte::fromIpUe(IPv4Datagram * datagram)
     }
 }
 
-void IP2lte::toStackUe(IPv4Datagram * datagram)
+void IP2lte::toStackUe(Packet * datagram)
 {
-    // obtain the encapsulated transport packet
-    cPacket * transportPacket = datagram->getEncapsulatedPacket();
-
     // 5-Tuple infos
     unsigned short srcPort = 0;
     unsigned short dstPort = 0;
-    int transportProtocol = datagram->getTransportProtocol();
-    // TODO Add support to IPv6
-    IPv4Address srcAddr  = datagram->getSrcAddress() ,
-                destAddr = datagram->getDestAddress();
+    // TODO Add support to Ipv6
```

Comment:
Take a look at L3Tools.cc and how the functions are used elsewhere, because it may help support both IPv4 and IPv6.

Reply:
Thanks for the pointer to the L3Tools. However, since there are many places where protocol support is hard-coded to IPv4 and the INET3 version of simulte is also hard-coded to IPv4, I would vote to add IPv6/more generic L3 support at a later point in time within a separate pull request. Added a comment with a reference to the L3Tools.